frontend/projects/upgrade/src/app/features/dashboard/feature-flags/pages/feature-flag-details-page/feature-flag-details-page-content/feature-flag-overview-details-section-card/feature-flag-overview-details-section-card.component.ts

```diff
@@ -61,9 +63,17 @@ export class FeatureFlagOverviewDetailsSectionCardComponent implements OnInit, O
     this.permissions$ = this.authService.userPermissions$;
     this.subscriptions.add(this.featureFlagService.currentUserEmailAddress$.subscribe((id) => (this.emailId = id)));
 
+    this.subscriptions.add(
```

Comment:
We can create an observable (or even better, an ngrx selector) that combines both 'permissions' and 'flag', we can make `menuButtonItems` itself an observable and simply capture the changes in the html. This general idea works in my local:

```
// component.ts:
this.flagAndPermissions$ = combineLatest([this.featureFlag$, this.permissions$]).pipe(
      map(([flag, permissions]) => ({ flag, permissions }))
    );

    this.menuButtonItems$ = this.flagAndPermissions$.pipe(
      map(({ flag, permissions }) => [
        { name: FEATURE_FLAG_DETAILS_PAGE_ACTIONS.EDIT, disabled: !permissions.featureFlags.update },
        { name: FEATURE_FLAG_DETAILS_PAGE_ACTIONS.DUPLICATE, disabled: !permissions.featureFlags.create },
        { name: FEATURE_FLAG_DETAILS_PAGE_ACTIONS.EXPORT_DESIGN, disabled: false },
        { name: FEATURE_FLAG_DETAILS_PAGE_ACTIONS.EMAIL_DATA, disabled: true },
        { name: FEATURE_FLAG_DETAILS_PAGE_ACTIONS.ARCHIVE, disabled: true },
        {
          name: FEATURE_FLAG_DETAILS_PAGE_ACTIONS.DELETE,
          disabled: !permissions?.featureFlags.delete || flag?.status === 'enabled',
        }
      ])
    );
    
 // html:
 [menuButtonItems]="menuButtonItems$ | async"
```